version: '3'
services:
  nginx:
    container_name: nginx
    build: ./nginx/
    image: nginx-transcendence
    env_file:
      - .env
    ports:
      - "80:80"
      - "443:443"
    depends_on:
      - django
    networks:
      - transcendence-net
    volumes:
      - ./nginx/www:/var/www/
    restart: on-failure

  django:
    container_name: django
    build: ./django
    image: django-transcendence
    env_file:
      - .env
    expose:
      - 8000
    depends_on:
      postgres:
        condition: service_healthy
    networks:
      - transcendence-net
    volumes:
      - ./django/transcendence:/var/src/app

  postgres:
    image: postgres:latest
    container_name: postgres
    env_file:
      - .env
    volumes:
      - postgres-vol:/var/lib/postgresql/data/
    networks:
      - transcendence-net
    restart: unless-stopped
    healthcheck:
      test: pg_isready -U ${POSTGRES_USER}
      interval: 1s
      retries: 20

  serverpong:
    container_name: serverpong
    build: ./serverPong
    image: django-serverpong
    restart: always
    depends_on:
      - django
    env_file:
      - .env
    expose:
      - 10000
    networks:
      - transcendence-net
    volumes:
      - ./serverPong/serverPong:/var/src/app

volumes:
  postgres-vol:
    name: postgres-vol
  nginx-vol:
    name: nginx-vol
  django-vol:
    name: django-vol

networks:
  transcendence-net:
    name: transcendence-net
    driver: bridge

